src/Utilities/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetAnalysis.PropertySetDataFlowOperationVisitor.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.PointsToAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Analyzer.Utilities.FlowAnalysis.Analysis.PropertySetAnalysis
+{
+    using PropertySetAnalysisData = IDictionary<AbstractLocation, PropertySetAbstractValue>;
+
+    internal partial class PropertySetAnalysis
+    {
+        /// <summary>
+        /// Operation visitor to flow the location validation values across a given statement in a basic block.
+        /// </summary>
+        private sealed class PropertySetDataFlowOperationVisitor :
+            AbstractLocationDataFlowOperationVisitor<PropertySetAnalysisData, PropertySetAnalysisContext, PropertySetAnalysisResult, PropertySetAbstractValue>
+        {
+            private const int MaxInterproceduralCallChain = 1;
+            private readonly ImmutableDictionary<IInvocationOperation, PropertySetAbstractValue>.Builder _hazardousUsageBuilderOpt;
+            private INamedTypeSymbol DeserializerTypeSymbol;
+
+            public PropertySetDataFlowOperationVisitor(PropertySetAnalysisContext analysisContext)
+                : base(analysisContext)
+            {
+                Debug.Assert(analysisContext.OwningSymbol.Kind == SymbolKind.Method);
```

Comment:
>analysisContext.OwningSymbol.Kind == SymbolKind.Method [](start = 29, length = 54)

I am presuming each of your analyzer is ensuring that you never call for code blocks such as field/property initializers? Regardless, you should add unit tests for each analyzer where there is a field/property initializer and verify the analyzer is not attempting to do flow-analysis.

Reply:
Will add unit tests, that assert might be a bad assumption.

Reply:
Added unit tests for a property setter and a field initializer.  Any other cases I should watch out for?  :-)